crates/flux-middle/src/global_env.rs

```diff
@@ -216,6 +223,44 @@ impl<'sess, 'tcx> GlobalEnv<'sess, 'tcx> {
         self.map().const_by_name(name)
     }
 
+    pub fn sort_of_assoc_pred(&self, impl_id: DefId, name: Symbol) -> Option<List<rty::Sort>> {
```

Comment:
The only error that's recoverable here is whether the `name` is a valid predicate, all other errors should be propagated up

```suggestion
    pub fn sort_of_assoc_pred(&self, impl_id: DefId, name: Symbol) -> QueryResult<Option<List<rty::Sort>>> {
```

If you have problems bridging between `QueryResult<T>` and `Result<T, ErrorGuaranteed>` you can use `.emit(genv.sess)` on a `QueryResult<T>` to convert it into a `Result<T, ErrorGuaranteed>`

Reply:
done.